src/modules/status.rs

```diff
@@ -57,6 +57,22 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     let segment_format = config.pipestatus_segment_format.unwrap_or(config.format);
 
     // Create pipestatus string
+    let pipestatus_separator =
+        match StringFormatter::new(config.pipestatus_separator).and_then(|formatter| {
+            formatter
+                .map_style(|variable| match variable {
+                    "style" => Some(Ok(config.style)),
+                    _ => None,
```

Comment:
I don't really like creating a separate `StringFormatter` for this. Maybe you could conditionally append the separator to the pipestatus format string?

Reply:
Sure, I've added a commit that removes the extra StringFormatter and instead adds the separator to each pipestatus format string except for the last one.

Reply:
Does the new version fit what you were suggesting?
https://github.com/starship/starship/pull/4264/files#diff-b377af853795c532ed1c949ed3014abe83237fb645587ff1681406d9fe726c0eR65-R71